### ISSUE TITLE
How to avoid default operator and default value setting

### ISSUE THREAD

**Issue by syama-aot** [react-querybuilder/react-querybuilder#305]
Hi,
its a awesome package that i have seen recently.In my application, based on selection of field, I want to select operator and after selection of operator , I want to show the value editor to select values. Is it possible to show only field select box (after adding a new rule ) without operator select box and value select box

**Comment by jakeboone02** [comment]
It seems like you're asking for two different things: 1) for new rules, only show the field selector at first (hide the operator selector and value editor), and 2) don't show the value editor until the user selects an operator.

The answer to (1) is yes: just use the `autoSelectField={false}` prop ([documentation](https://react-querybuilder.js.org/docs/api/querybuilder#autoselectfield), [demo](https://react-querybuilder.js.org/react-querybuilder/#autoSelectField=false)).

For (2) you'd probably need to use a custom `ValueEditor` that hides itself when operator is some default value like `'~'` (that's how `autoSelectField={false}` works).

Can you confirm which one you're asking for? Or is it both?

**Comment by jakeboone02** [comment]
Actually it was pretty easy to implement both techniques, so I created [this codesandbox](https://codesandbox.io/s/react-query-builder-dependent-value-editor-mpmxg5?file=/src/App.tsx) to demonstrate. Here is the code:

```tsx
import { useState } from 'react';
import { defaultOperators, Field, NameLabelPair, QueryBuilder, RuleGroupType, ValueEditor, ValueEditorProps } from 'react-querybuilder';
import 'react-querybuilder/dist/query-builder.scss';

const getOperators = (): NameLabelPair[] => [
  { name: '~', label: '------' },
  ...defaultOperators,
];

const fields: Field[] = [
  { name: 'firstName', label: 'First Name' },
  { name: 'lastName', label: 'Last Name' },
];

const CustomValueEditor = (props: ValueEditorProps) => {
  if (props.operator === '~') {
    return null;
  }
  return <ValueEditor {...props} />;
};

export default function App() {
  const [query, setQuery] = useState<RuleGroupType>({
    combinator: 'and',
    rules: [],
  });

  return (
    <div>
      <QueryBuilder
        fields={fields}
        query={query}
        onQueryChange={(q) => setQuery(q)}
        autoSelectField={false}
        getOperators={getOperators}
        getDefaultOperator="~"
        controlElements={{
          valueEditor: CustomValueEditor,
        }}
      />
    </div>
  );
}
```

**Comment by jakeboone02** [comment]
I got inspired by your request and actually implemented a new prop called `autoSelectOperator` that works like `autoSelectField` but for operators (see #306). So, in the next release, all you'll have to do is this:

```tsx
<QueryBuilder
  fields={fields}
  query={query}
  onQueryChange={(q) => setQuery(q)}
  autoSelectField={false}
  autoSelectOperator={false} // <-- this will be available in react-querybuilder@4.3.0
/>
```

**Comment by syama-aot** [comment]
Thank you very much

You are awesome.
